packages/cli/src/modules/insights/database/repositories/insights-by-period.repository.ts

```diff
@@ -376,4 +376,13 @@ export class InsightsByPeriodRepository extends Repository<InsightsByPeriod> {
 
 		return aggregatedInsightsByTimeParser.parse(rawRows);
 	}
+
+	async pruneOldData(maxAgeInDays: number): Promise<{ affected: number }> {
+		const result = await this.createQueryBuilder()
+			.delete()
+			.where(`${this.escapeField('periodStart')} < ${this.getPeriodFilterExpr(maxAgeInDays)}`)
+			.execute();
+
+		return { affected: result.affected ?? 0 };
```

Comment:
If `affected` is nullish it means nothing was deleted? Or something went wrong?

Reply:
affected is null or undefined if the sql driver does not return it. I changed to code to return the nullable value, and log it only if set